src/sonic-build-hooks/scripts/buildinfo_base.sh

```diff
@@ -13,6 +13,7 @@ WEB_VERSION_FILE=$VERSION_PATH/versions-web
 BUILD_WEB_VERSION_FILE=$BUILD_VERSION_PATH/versions-web
 REPR_MIRROR_URL_PATTERN='http:\/\/packages.trafficmanager.net\/debian'
 DPKG_INSTALLTION_LOCK_FILE=/tmp/.dpkg_installation.lock
+GET_RETRY_COUNT=5
```

Comment:
why this one is 5 while ones are 20?

Reply:
It is not really 5 retries. By default, wget retries 20 times, see this [man page](https://linux.die.net/man/1/wget):
```
--tries=number
Set number of retries to number. Specify 0 or inf for infinite retrying. The default is to retry 20 times, with the exception of fatal errors like "connection refused" or "not found" (404), which are not retried.
```
20 retries weren't enough, so we run it 5 times in the loop.

For the curl we add "--retry" parameter, cause it doesn't retry by default, see [man page](https://linux.die.net/man/1/curl):
```
--retry <num>
If a transient error is returned when curl tries to perform a transfer, it will retry this number of times before giving up. Setting the number to 0 makes curl do no retries (which is the default).
```
To retry even "connection refused" we add "--retry-connerefused" option, which is shared between curl and wget.

Reply:
Is 20x5 retries too many? Since there is x2 backoff in curl retry, it will take a long time to reach just 20. So it is not considered as transient issue.

Reply:
My question is still applicable.

Reply:
It is not. 

> Is 20x5 retries too many? Since there is x2 backoff in curl retry, it will take a long time to reach just 20. So it is not considered as transient issue.

it is not. When I saw instability the default behavior was failing in less than a minute, maybe 20-30 seconds. So in total, it will take maybe 2-3 minutes. IMO it is acceptable

Reply:
Both `curl` and `wget` have their own retry parameters. I do not understand the purpose of another for-loop outside the command lines.
1. `wget` supports linear backoff
2. `curl` supports exponential backoff (double each time)

With proper command parameter, you can achieve retry in long time, like 2~3 min as you intend.

Reply:
I will describe the problem I faced: I had to connect to a very unstable server, which sometimes (but pretty frequently) got hanging.
I played with the parameters for both commands and neither of them helped me, it was still failing on broken connection, etc. The only thing that helped me to completely avoid all the server instabilities is adding that loop. In an ideal world, it doesn't add any overhead, since it will work on the first iteration.

Reply:
Could you explore below functionality?
1. wget supports linear backoff
2. curl supports exponential backoff (double each time)
Tuning these parameters will be easier than for-loop. in this PR.

Reply:
> Could you explore below functionality?
> 
> 1. wget supports linear backoff
> 2. curl supports exponential backoff (double each time)
>    Tuning these parameters will be easier than for-loop. in this PR.

@qiluo-msft I am using them as well. See `src/sonic-build-hooks/hooks/curl`. I tried to avoid that loop and I used `--tries=100` for `wget` and `--retry 25` for `curl` and it didn't help me to avoid such failure I mentioned in the description. Seems like only a complete retry of`wget`/`curl` works. The only thing that helps me is to run it in the loop I added. So I insist on adding it.